Level 2

ReverseSerj opened the pull request:
Приветствую вас. Не знаю, на сколько критичен пиксель пёрфект футора, по роуту '/saved-movies', но я так и не нашёл решения, как сделать его максимально приближенным, на остальных вроде всё в норме.

ReverseSerj committed:
layout

ReverseSerj committed:
Update README.md

nehydna commented:
Например,# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- ✅ <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета. 
-  <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.

На всех страницах на всем адаптиве блоки и элементы не должны выходить за пределы контейнеров. Например, http://joxi.ru/BA0YQZ1IqDKGzr, но надо исправить для всех похожих случаев

- ✅ <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить
-  <!--5.55--> Именование классов сделаны по БЭМ

Необходимо использовать БЭМ-нотацию. В проекте есть элементы без блоков. Проверить БЭМ можно тут: https://yoksel.github.io/html-tree/

-  <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях

На всем адаптиве на всех страницах блоки и элементы не должны прилипать к краям контейнера. Например, http://joxi.ru/v29YR1XIjykVJ2. Надо исправить для всех похожих случаев
На всем адаптиве контент должен быть отценрован http://joxi.ru/n2YJL1ZC0zdXvA

-  <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, обертки не состоят из див и т.д.)

Между шапкой и подвалом должна быть расположена главная часть страницы. 
Ее надо разметить семантичным тегом main. Все секции должны быть внутри. Надо исправить для всех страниц.

- ✅ <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- ✅ <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- ✅ <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- ✅ <!--3.38--> Работа модальных окон настроена.
- ✅ Разметка портирована в JSX:
	- ✅ <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- ✅ <!--3.38--> разметка вынесена в соответствующие ей компоненты

-  В проекте есть:
	- ✅ <!--3.38--> директория images/ с картинками,
            - ✅ <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- ✅ <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
-  <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.


Ссылки и кнопки должны менять прозрачность при наведении. Например, http://joxi.ru/L21B5aXUDMZ5DA (тут еще и неверно прописано значение свойства). Необходимо задать наведение для всех интерактивных элементов. Необходимо свериться с UI-кит


- ✅  <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- ✅  <!--2.14--> Шрифты подключены через @font-face.
- ✅ <!--2.14--> Иконки экспортированы в формате SVG.
-  <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение

Поля не выделяются

-  <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля

Плейсхолдеров нет, поля не валидированы

- ✅ <!--2.14--> reset.css запрещен.

### Рекомендации
- ✅ <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- ✅ <!--1.67--> Для изображений задан атрибут alt с подходящим значением.
- ✅ <!--1.67--> Растровые и векторные изображения оптимизированы;


## Количество баллов: 66
## Доп. Комментарии
- Если есть что-то, что не вошло в чеклист
Неверно указан атрибут lang

ReverseSerj committed:
work on mistakes

change lang, add style state submit button(disabled), focus input, validation form, screen resolution adaptation, add navigation active state and  block 'main' on all pages,

ReverseSerj committed:
Merge branch 'level-2' of https://github.com/ReverseSerj/movies-explorer-frontend into level-2

ReverseSerj commented:
Приветсвую. Не нашёл проблемы с opacity кнопки "назад" по роуту '*' .

ReverseSerj committed:
fixes

fix background 404 and adaptive resolution layout component 'Profile'

nehydna commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- ✅ <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета. 
- ✅ <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- ✅ <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить
-  <!--5.55--> Именование классов сделаны по БЭМ

Необходимо использовать БЭМ-нотацию.  Проверить БЭМ можно тут: https://yoksel.github.io/html-tree/


- ✅ <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях
- ✅ <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, обертки не состоят из див и т.д.)
- ✅ <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- ✅ <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- ✅ <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- ✅ <!--3.38--> Работа модальных окон настроена.
- ✅ Разметка портирована в JSX:
	- ✅ <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- ✅ <!--3.38--> разметка вынесена в соответствующие ей компоненты

-  В проекте есть:
	- ✅ <!--3.38--> директория images/ с картинками,
            - ✅ <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- ✅ <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
-  <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.

Ссылки и кнопки должны менять прозрачность при наведении. Например, http://joxi.ru/BA0YQZ1IqnqLor, но надо исправить для всех интерактивных элементов

- ✅  <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- ✅  <!--2.14--> Шрифты подключены через @font-face.
- ✅ <!--2.14--> Иконки экспортированы в формате SVG.
-  <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение

Поля выделяются не везде http://joxi.ru/1A5YLnXIwXwe6A

-  <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля

Подсказок нет

- ✅ <!--2.14--> reset.css запрещен.

### Рекомендации
- ✅ <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- ✅ <!--1.67--> Для изображений задан атрибут alt с подходящим значением.
- ✅ <!--1.67--> Растровые и векторные изображения оптимизированы;


## Количество баллов: 88
## Доп. Комментарии
- Если есть что-то, что не вошло в чеклист
